### PR TITLE
Add fastjet, hepmc, yoda, rivet, lhapdf, herwig, pythia, sacrifice, nlojet++

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

This provides packages for several high energy physics packages, including most known Pythia and Herwig codes. The additions are presented together because of dependencies between packages.

The derivations are mostly done in a standard way, with two exceptions. Rivet uses texlive to render it's plots which brings some minor complications with paths. Also rivet-buildanalysis utility is supposed to invoke the same compiler that rivet was compiled with, so there is a dirty hack to call the cc-wrapper in runtime. The other package, lhapdf, is designed to install it's data files into $prefix/usr/share, which is not a nix way approach. A nix expression to install these data files to a separate store paths is provided. `preferLocalBuild = true` prevents the data files from being cached with Hydra.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [x] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/physics/rivet/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{ stdenv, fetchurl, fastjet, ghostscript, gsl, hepmc, imagemagick, less, python, texlive, yoda, which, makeWrapper }:
```
Comment (Mic92) [inline pkgs/development/libraries/physics/rivet/default.nix]: we soon want to make python3 default. Could already set python to python2 here?

**Comment by Mic92** [comment]
looks good otherwise

File: pkgs/development/libraries/physics/fastjet/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  configureFlags = "--enable-allcxxplugins";
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "A software package for jet finding in pp and e+e− collisions";
+    license     = stdenv.lib.licenses.gpl2;
+    homepage    = http://fastjet.fr/;
+    platforms   = stdenv.lib.platforms.unix;
```
Comment (veprbl) [inline pkgs/development/libraries/physics/fastjet/default.nix]: Effectively, yes. But what is the advantage of having my name and email in meta.maintainers? Will I get notifications from hydra? For the rest we rely on mentions-bot anyway.

**Comment by Mic92** [comment]
Thanks!

**Comment by veprbl** [comment]
Thank you for reviewing this!

**Pull request merged**